financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheetViewModel.kt

```diff
@@ -258,29 +286,79 @@ internal class FinancialConnectionsSheetViewModel @Inject constructor(
+                            onStartApp2App(receivedUrl.toString()
+                                .replaceFirst("stripe-auth://native-redirect/$applicationId/", ""))
+
+                        // stripe-auth://link-accounts/.../authentication_return
+                        receivedUrl?.host == "link-accounts" && receivedUrl?.buildUpon()?.clearQuery()?.build()?.path == "/$applicationId/authentication_return" ->
+                            onReturnUrlReceived(receivedUrl)
+
+                        // stripe-auth://link-accounts/{applicationId/success
+                        receivedUrl?.buildUpon()?.clearQuery()
+                            .toString() == state.manifest?.successUrl -> onFlowSuccess(state, receivedUrl)
+
+                        // stripe-auth://link-accounts/{applicationId/cancel
+                        receivedUrl?.buildUpon()?.clearQuery()
+                            .toString() == state.manifest?.cancelUrl -> onFlowCancelled(state)
+                        
+                        else -> {
+                            setState { copy(webAuthFlowStatus = AuthFlowStatus.NONE) }
+                            onFatal(
+                                state,
+                                Exception("Error processing FinancialConnectionsSheet intent")
+                            )
+                        }
                     }
-                    state.manifest?.cancelUrl -> onUserCancel(state)
-                    else ->
-                        onFatal(
-                            state,
-                            Exception("Error processing FinancialConnectionsSheet intent")
-                        )
                 }
             }
         }
     }
 
+    private fun onStartApp2App(unwrappedUriString: String) {
+        setState {
+            copy(
+                webAuthFlowStatus = AuthFlowStatus.APP2APP,
+                viewEffect = OpenAuthFlowWithUrl(unwrappedUriString)
+            )
+        }
+    }
+
+    private  fun onFlowSuccess(state: FinancialConnectionsSheetState, receivedUrl: Uri?) {
+        if (receivedUrl == null) {
+            onFatal(state, Exception("Intent url received from web flow is null"))
+        }
+        setState { copy(webAuthFlowStatus = AuthFlowStatus.NONE) }
+        when (state.initialArgs) {
+            is ForData -> fetchFinancialConnectionsSession(state)
+            is ForToken -> fetchFinancialConnectionsSessionForToken(state)
+            is ForLink -> onSuccessFromLinkFlow(receivedUrl)
+        }
+    }
+
+    private fun onFlowCancelled(state: FinancialConnectionsSheetState) {
+        setState { copy(webAuthFlowStatus = AuthFlowStatus.NONE) }
+        onUserCancel(state)
+    }
+
+    private fun onReturnUrlReceived(receivedUrl: Uri) {
+        setState {
+            val authFlowResumeUrl =
+                "${manifest!!.hostedAuthUrl}&startPolling=true&${receivedUrl.fragment}"
+            copy(
+                webAuthFlowStatus = AuthFlowStatus.APP2APP,
```

Comment:
@carlosmuvi-stripe should this state be `AuthFlowStatus.WEB` instead? Since we are coming back from app2app to the webflow?

Reply:
It should be `APP2APP`. `onResume` The whole purpose of it is to avoid closing the SDK when `onResume` gests called right after this method due to the Android lifecycle. 

When `onResume` gets called, we inmediately move from `APP2APP` to `WEB`.

https://github.com/stripe/stripe-android/blob/98cd83b92443ff4d90b1866131c515642824d369/financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheetViewModel.kt#L169

Reply:
more context on Android lifecycle in the SDK - https://docs.google.com/document/d/1W0fSk9sLrNrPPLwJRH7r_Z_Eu6XraD2GYB1Z-qB33Ag/edit (though it's not having App2App into account)